src/main/java/org/antlr/intellij/plugin/ANTLRv4ASTFactory.java

```diff
@@ -29,6 +29,7 @@ public class ANTLRv4ASTFactory extends ASTFactory {
 		ruleElementTypeToPsiFactory.put(ANTLRv4TokenTypes.RULE_ELEMENT_TYPES.get(ANTLRv4Parser.RULE_modeSpec), ModeSpecNode.Factory.INSTANCE);
 		ruleElementTypeToPsiFactory.put(ANTLRv4TokenTypes.RULE_ELEMENT_TYPES.get(ANTLRv4Parser.RULE_action), AtAction.Factory.INSTANCE);
 		ruleElementTypeToPsiFactory.put(ANTLRv4TokenTypes.RULE_ELEMENT_TYPES.get(ANTLRv4Parser.RULE_identifier), TokenSpecNode.Factory.INSTANCE);
+		ruleElementTypeToPsiFactory.put(ANTLRv4TokenTypes.RULE_ELEMENT_TYPES.get(ANTLRv4Parser.RULE_tokensSpec), TokensSpecNode.Factory.INSTANCE);
```

Comment:
I'm not sure we need this, token definitions already have a parent of type `TokenSpecNode`:
<img width="912" alt="image" src="https://github.com/antlr/intellij-plugin-v4/assets/281528/9eb96e69-e26c-4fe5-8433-bdb0e3d566c1">
So using `getParent() instanceof TokenSpecNode` should be enough.